lib/cli-engine.js

```diff
@@ -744,12 +874,16 @@ class CLIEngine {
      * This is the same logic used by the ESLint CLI executable to determine
      * configuration for each file it processes.
      * @param {string} filePath The path of the file to retrieve a config object for.
-     * @returns {Object} A configuration object for the file.
+     * @returns {ConfigData} A configuration object for the file.
      */
-    getConfigForFile(filePath) {
-        const configHelper = this.config;
-
-        return configHelper.getConfig(filePath);
+    getConfigForFile(filePath = "a.js") {
```

Comment:
```js
    getConfigForFile(filePath = "a.js") {
        const { configArrayFactory, options } = internalSlotsMap.get(this);
        const absolutePath = path.resolve(options.cwd, filePath);
```

Because:

- that `path.resolve` requires string arguments, 
- there are some tests that call `getConfigForFile()` method with no arguments, 
- and I felt odd to use a directory path to test `overrides` matching.

However, indeed, this will change `overrides` resolving when no arguments. I will make it back.